pkg/ovs/openflow/ofctrl_bridge_test.go

```diff
@@ -160,3 +166,75 @@ func TestOFBridgePacketRcvd(t *testing.T) {
 		}
 	}
 }
+
+func TestOFMeterStats(t *testing.T) {
+	// Reset OVS metrics (Prometheus) and reinitialize them to test.
+	legacyregistry.Reset()
+	metrics.InitializeOVSMetrics()
+
+	b := NewOFBridge("test-br", GetMgmtAddress(ovsconfig.DefaultOVSRunDir, "test-br"))
+	sw := newFakeOFSwitch(b)
+	b.SetOFSwitch(sw)
+
+	mpMeterStatsReply := openflow15.NewMpReply(openflow15.MultipartType_MeterStats)
+	meterStats1 := openflow15.NewMeterStats(1)
+	stats1 := openflow15.NewMeterBandStats()
+	stats1.PacketBandCount = 100
+	stats1.ByteBandCount = 200
+	meterStats1.AddBandStats(*stats1)
+	meterStats2 := openflow15.NewMeterStats(2)
+	stats2 := openflow15.NewMeterBandStats()
+	stats2.PacketBandCount = 0
+	stats2.ByteBandCount = 0
+	meterStats2.AddBandStats(*stats2)
+	// Unexpected case
+	meterStats3 := openflow15.NewMeterStats(3)
+	stats3 := openflow15.NewMeterBandStats()
+	stats3.PacketBandCount = 300
+	stats3.ByteBandCount = 400
+	meterStats3.AddBandStats(*stats3)
+	mpMeterStatsReply.Body = append(mpMeterStatsReply.Body, meterStats1)
+	mpMeterStatsReply.Body = append(mpMeterStatsReply.Body, meterStats2)
+	mpMeterStatsReply.Body = append(mpMeterStatsReply.Body, meterStats3)
+	expectedOVSMeterPacketCount := `
+	# HELP antrea_agent_ovs_meter_packet_dropped_count [ALPHA] Number of packets dropped by OVS meter. The value is more than 0 when the packets exceed the rate-limit.
+	# TYPE antrea_agent_ovs_meter_packet_dropped_count gauge
+   antrea_agent_ovs_meter_packet_dropped_count{meter_id="PacketInMeterNetworkPolicy"} 100
+   antrea_agent_ovs_meter_packet_dropped_count{meter_id="PacketInMeterTraceflow"} 0
+	`
+	handleMeterStatsReply := func(packetCounts map[int]int64) {
+		for k, v := range packetCounts {
+			switch k {
+			case 1:
+				metrics.OVSMeterPacketDroppedCount.WithLabelValues("PacketInMeterNetworkPolicy").Set(float64(v))
+			case 2:
+				metrics.OVSMeterPacketDroppedCount.WithLabelValues("PacketInMeterTraceflow").Set(float64(v))
+			default:
+				klog.InfoS("Received unexpected meterId", "meterId", k)
```

Comment:
should this be `assert.Fail` instead of `klog.InfoS`?

Reply:
This one does not need to throw errors according to my original design. I am thinking that we can just ignore it if we receive an unexpected id, so the function here is basically consistent with the one defined in `func (c *client) GetMeterStats()`